### DISCUSSION TITLE
Environment Variables

### DISCUSSION THREAD

**Discussion by CruzanCaramele** [fastapi/typer#552] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.command(short_help="Creates Secrets")
def create(
    name: str = typer.Option(None, "--token-name", help="The name of the token."),
    auth: str = typer.Option(
        None,
        "--auth",
        envvar="AUTH_TOKEN",
        help=".....",
    ),
    config: Optional[Path] = typer.Option(
        None,
        exists=True,
        readable=True,
        file_okay=True,
        resolve_path=True,
    ),
):
```

### Description

In the above example function definition, is there a way to make the argument environment variable **auth**    key **envvar="AUTH_TOKEN"** to be defined by the user instead, rather than hardcoding **AUTH_TOKEN**  ?

### Operating System

Windows

### Typer Version

0.7.0

### Python Version

Python 3.11.0

**Comment by jayqi** [comment, marked as answer]
Have you considered changing your command's API to take an environment variable name as an option and then read from that environment variable internal to the command? That seems like the clearest way to do things. I don't think you could otherwise make one `--auth` option mean two different things cleanly.
